### PR TITLE
digital: qa: Add preconditions to wait for qa_packet_headerparser_b

### PR DESCRIPTION
**Pull request by mbr0wn**
This is an attempt to avoid QA failures due to messages not being
delivered before wait() concludes.

### CODE REVIEW

File: gr-digital/python/digital/qa_packet_headerparser_b.py [hunk 1]
```diff
@@ -16,6 +16,16 @@
 import pmt
 
 
+def _wait_for_msgs(precondition):
+    """
+    Simple loop to check for a condition
+    """
+    stop_time = time.time() + 2.0
```
Comment (jdemel) [inline gr-digital/python/digital/qa_packet_headerparser_b.py:23]: Maybe put the default time `2.0` in a function parameter.

File: gr-digital/python/digital/qa_packet_headerparser_b.py [hunk 2]
```diff
@@ -16,6 +16,16 @@
 import pmt
 
 
+def _wait_for_msgs(precondition):
```
Comment (jdemel) [inline gr-digital/python/digital/qa_packet_headerparser_b.py:19]: Would it make sense to add this function as a method to `gr_unittest`?

**Review by jdemel — COMMENTED** [review]
I have 2 suggestions for broader application. I assume these would make msg QA tests better for everyone. And easier to use.

**Comment by mormj** [comment]
@mbr0wn - does the suggestion to move to gr_unittest seem reasonable?

**Comment by mbr0wn** [comment]
Yeah, I think it is.

File: gr-digital/python/digital/qa_packet_headerparser_b.py (hunk 2 above)
Comment (mbr0wn) [reply gr-digital/python/digital/qa_packet_headerparser_b.py]: OK done.

File: gr-digital/python/digital/qa_packet_headerparser_b.py (hunk 1 above)
Comment (mbr0wn) [reply gr-digital/python/digital/qa_packet_headerparser_b.py]: It is in gr_unittest

**Review by mormj — APPROVED** [review]

**Comment by mormj** [comment]
Feedback has been addressed.  Merging.

**Pull request merged**